packages/brand-positioning-block/src/components/HorizontalAxis.tsx

```diff
@@ -0,0 +1,60 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useGuidelineDesignTokens } from '@frontify/guideline-blocks-shared';
+import { useEffect, useRef, useState } from 'react';
+import { AxisProps } from '../types';
+
+export const HorizontalAxis = ({}: AxisProps) => {
+    const { designTokens } = useGuidelineDesignTokens();
+    const [labelWidth, setLabelWidth] = useState<string | number>('auto');
+
+    const ref = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        /**
+         * Labels are rotated 90 degrees on mobile screen.
+         * This is needed to calculate the new width of the labels after rotated.
+         */
+        const handleWindowResize = () => {
+            if (window.innerWidth <= 600 && ref.current) {
+                const { width } = ref.current.getBoundingClientRect();
+                setLabelWidth(width);
+            } else {
+                if (labelWidth !== 'auto') {
+                    setLabelWidth('auto');
+                }
+            }
+        };
+        handleWindowResize();
+        window.addEventListener('resize', handleWindowResize);
```

Comment:
It would be great to handle the styles with CSS. Something we can do once we figure out the functionality

Reply:
Actually it is very tricky. Rotating 90 degrees does not change the width of the element, and it would take a lot of space up on mobile. If you change the label to absolute positioning the width does change after rotation, however then it is removed from the document flow.

I spent some time trying to come up with a CSS only solution and failed to do so. I do realise this is a bit ugly, so if you would have some ideas, POCs, codepens, I would highly appreciate it.

Reply:
Done